test/Salsa.jl

```diff
@@ -48,17 +48,16 @@ Salsa.@component Workspace begin
     Salsa.@input i2::Salsa.InputScalar{Int}
     Salsa.@connect compiler::Compiler
 end
+Salsa.@component TestStorage begin
+    Salsa.@connect workspace::Workspace
```

Comment:
Please document and explain what `@connect` does. If/why shouldn't someone use `@connect` and how can they achieve the same goal (similar to ExecutionGraph/Workspace use-case).

Reply:
It's in the docstring for `@connect` already! :)  I've just added a paragraph that hopefully explains this a bit more. :)